server.js

```diff
@@ -1,15 +1,256 @@
+const requestProxy = require('express-request-proxy');
 const PORT = process.env.PORT || 3000;
+const app = express();
 
+const connectionString = process.env.DATABASE_URL;
+const client = new pg.Client(connectionString);
+client.connect();
+client.on('error', err => console.error(err));
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
 app.use(express.static('./public'));
 
-app.get('*', function(request, response) {
-  response.sendFile('./public/index.html', { root : __dirname });
+app.get('/', function(request, response) {
+  response.sendFile('./public/index.html', {root: './public'});
+})
+
+app.get('/snake', function(request, response) {
+  response.sendFile('./public/index.html', {root: './public'});
+})
+
+app.get('/about', function(request, response) {
+  response.sendFile('./public/index.html', {root: './public'});
 })
 
+app.get('/project', function(request, response) {
+  client.query(`
+    SELECT * FROM project p
+    INNER JOIN project_technology pt
+      ON p.project_id = pt.project_id;`
+  )
+  .then(function(result) {
+    var groupBy = result.rows.reduce(function(acc,cur) {
+      let project = acc.find(v => v.project_id === cur.project_id);
+
+      if (project === undefined) {
+        project = cur;
+        project.technologies = [];
+        acc.push(project);
+      }
+
+      project.technologies.push(cur.technology_id);
+
+      delete cur.technology_id;
+
+      return acc;
+    }, []);
+
+    response.send(groupBy);
+  })
+  .catch(console.error);
+})
+
+app.get('/technology', function(request, response) {
+  client.query(`
+    SELECT * FROM technology;`
+  )
+  .then(function(result) {
+    response.send(result);
+  })
+  .catch(console.error);
+})
+
+app.get('/technology/:id', (request, response) => {
+  client.query(`
+    SELECT * FROM technology
+    WHERE technology_id = ${request.params.id};`
+  )
+  .then(result => response.send(result.rows))
+  .catch(console.error);
+});
+
+app.get('/github/*', proxyGitHub);
+
+app.get('*', function(request,response) {
+  response.status(404).send('That page does not exist.');
+});
+
+function proxyGitHub(request, response) {
+  console.log('Routing GitHub request for', request.params[0]);
+  (requestProxy({
+    url: `https://api.github.com/${request.params[0]}`,
+    headers: {
+      Authorization: `token ${process.env.GITHUB_TOKEN}`
+    }
+  }))(request, response);
+}
+
 app.listen(PORT, function() {
   console.log('Listening on port: ', PORT);
-});
\ No newline at end of file
+});
+
+let projectsData;
+
+getLocalData();
+
+function getLocalData() {
+  fs.readFile('./public/data/projects.json', function (err, localProjectsData) {
+    if (err) processError(err);
+    processLocalData(JSON.parse(localProjectsData));
+    getRemoteData();
+  });
+}
+
+function processError(err) {
+  console.error(err);
+}
+
+function processLocalData(localProjectsData) {
+  projectsData = localProjectsData;
+}
+
+function getRemoteData() {
```

Comment:
You need to use a proxy to make this request secure. You should make an AJAX request to your `/github/*`  route which already has the proxy set up.